test_unsharp_masking_output_type_and_shape fails on non-x86 architectures

### Description:

When building the Debian package, the **test_unsharp_masking_output_type_and_shape** and **test_pyramid_dtype_support** tests fail for many non-x86 architectures because of invalid value runtime warnings:
```
_ test_unsharp_masking_output_type_and_shape[True--1.0--1.0-2.0-uint64-shape2-False] _
[…]
    def test_unsharp_masking_output_type_and_shape(
        radius, amount, shape, multichannel, dtype, offset, preserve
    ):
        array = np.random.random(shape)
>       array = ((array + offset) * 128).astype(dtype)
E       RuntimeWarning: invalid value encountered in cast

skimage/filters/tests/test_unsharp_mask.py:41: RuntimeWarning
```
and
```
_____________ test_pyramid_dtype_support[pyramid_laplacian-uint8] ______________

pyramid_func = <function pyramid_laplacian at 0xffff78ab8a40>, dtype = 'uint8'

    @pytest.mark.parametrize('dtype', ['float16', 'float32', 'float64', 'uint8', 'int64'])
    @pytest.mark.parametrize(
        'pyramid_func', [pyramids.pyramid_gaussian, pyramids.pyramid_laplacian]
    )
    def test_pyramid_dtype_support(pyramid_func, dtype):
>       img = np.random.randn(32, 8).astype(dtype)
E       RuntimeWarning: invalid value encountered in cast

skimage/transform/tests/test_pyramids.py:197: RuntimeWarning
```
While I can (and will) ignore the warnings locally for the build, it may be useful to handle this upstream.

### Way to reproduce:

See above

### Version information:

```Shell
numpy 1.26.4
Python 3.11 and 3.12
Debian unstable
skimage 0.23.1

So far on arm64, armhf, ppc64el, s390x.
```
All build logs [here](https://buildd.debian.org/status/logs.php?pkg=skimage&ver=0.23.1-1&suite=sid).

--- a/skimage/filters/tests/test_unsharp_mask.py
+++ b/skimage/filters/tests/test_unsharp_mask.py
@@ -1,3 +1,5 @@
+import warnings
+
 import numpy as np
 import pytest
 
@@ -38,7 +40,17 @@ def test_unsharp_masking_output_type_and_shape(
     radius, amount, shape, multichannel, dtype, offset, preserve
 ):
     array = np.random.random(shape)
-    array = ((array + offset) * 128).astype(dtype)
+    array = (array + offset) * 128
+    with warnings.catch_warnings():
+        # Ignore arch specific warning on arm64, armhf, ppc64el, riscv64, s390x
+        # https://github.com/scikit-image/scikit-image/issues/7391
+        warnings.filterwarnings(
+            action="ignore",
+            category=RuntimeWarning,
+            message="invalid value encountered in cast",
+        )
+        array = array.astype(dtype)
+
     if (preserve is False) and (dtype in [np.float32, np.float64]):
         array /= max(np.abs(array).max(), 1.0)
     channel_axis = -1 if multichannel else None

--- a/skimage/transform/tests/test_pyramids.py
+++ b/skimage/transform/tests/test_pyramids.py
@@ -1,4 +1,5 @@
 import math
+import warnings
 
 import pytest
 import numpy as np
@@ -194,8 +195,16 @@ def test_check_factor():
     'pyramid_func', [pyramids.pyramid_gaussian, pyramids.pyramid_laplacian]
 )
 def test_pyramid_dtype_support(pyramid_func, dtype):
-    img = np.random.randn(32, 8).astype(dtype)
-    pyramid = pyramid_func(img)
+    with warnings.catch_warnings():
+        # Ignore arch specific warning on arm64, armhf, ppc64el, riscv64, s390x
+        # https://github.com/scikit-image/scikit-image/issues/7391
+        warnings.filterwarnings(
+            action="ignore",
+            category=RuntimeWarning,
+            message="invalid value encountered in cast",
+        )
+        img = np.random.randn(32, 8).astype(dtype)
 
+    pyramid = pyramid_func(img)
     float_dtype = _supported_float_type(dtype)
     assert np.all([im.dtype == float_dtype for im in pyramid])

